From 33c710c212f9ae8a34960a2ba1f313b992526027 Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Thu, 8 Aug 2024 00:02:31 -0400
Subject: [PATCH] Fuck pdm and prepare for distro packaging

Some motherfucker really fancies this "modern" build tool pdm.
"pdm" pulls in 20+ dependencies and requires bootstrapping - yet no
meaningful features I can see anywhere for distro maints.

Whoever thinks PEP517 with this pdm shit is doing any good should pay a
visit to a mental health doctor. Stop it, get some help.

Signed-off-by: Tianhao Chai <cth451@gmail.com>
---
 pyproject.toml | 53 +++++---------------------------------------------
 1 file changed, 5 insertions(+), 48 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 27455176..f3086d93 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -39,8 +39,8 @@ Homepage = "https://github.com/Backblaze/b2-sdk-python"
 hook-dirs = "b2sdk._pyinstaller:get_hook_dirs"
 
 [build-system]
-requires = ["pdm-backend"]
-build-backend = "pdm.backend"
+requires = ["setuptools>=43.0.0", "wheel", "setuptools_scm>=7"]
+build-backend = "setuptools.build_meta"
 
 [tool.liccheck]
 authorized_licenses = [
@@ -140,50 +140,7 @@ directory = "infrastructure"
 name = "Infrastructure"
 showcontent = true
 
-[tool.pdm]
-package-type = "library"
+[tool.setuptools.packages.find]
+include = ["b2sdk", "b2sdk.*"]
 
-[tool.pdm.build]
-includes = ["b2sdk"]
-
-[tool.pdm.version]
-source = "scm"
-
-[tool.pdm.dev-dependencies]
-format = [
-    "yapf==0.27",
-    "ruff==0.1.15",
-]
-lint = [
-    "yapf==0.27",
-    "ruff==0.1.15",
-    "pytest==6.2.5",
-    "liccheck==0.9.2",
-    "setuptools",
-]
-test = [
-    "coverage==7.2.7",
-    "pytest==6.2.5",
-    "pytest-cov==3.0.0",
-    "pytest-mock==3.6.1",
-    "pytest-lazy-fixture==0.6.3",
-    "pytest-xdist==2.5.0",
-    "pytest-timeout==2.1.0",
-    "tqdm<5.0.0,>=4.5.0",
-    "eval_type_backport>=0.1.3,<1; python_version<'3.10'",  # used with pydantic
-    # remove `and platform_python_implementation!='PyPy'` after dropping Python 3.7 support as that
-    # will allow us to update pydantic to a version which installs properly under PyPy
-    "pydantic>=2.0.1,<3; python_version>='3.8' and platform_python_implementation!='PyPy'",
-]
-release = [
-    "towncrier==23.11.0; python_version>='3.8'",
-]
-doc = [
-    "sadisplay>=0.4.9",
-    "sphinx>=5.3.0, <6",
-    "sphinx-autobuild>=2021.3.14",
-    "sphinx-rtd-theme>=2.0.0",
-    "sphinx-autodoc-typehints>=1.23.0",
-    "sphinxcontrib-plantuml>=0.27",
-    "tornado>=6.3.3; python_version>='3.8'",
-]
+[tool.setuptools_scm]
-- 
2.46.0

